lint-rules.md

```diff
@@ -0,0 +1,1280 @@
+
+```wdl
+Boolean complicated_logic = (
+    if !(
+        a && b || c && (!d || !e)
+        == (
+            if foobar
+            then gak
+            else caz
+        )
+    )
+    then "wow"
+    else "WOWOWOW"
+)
+```
+
+The same expression with as many newlines inserted as permissible:
+
+```wdl
+Boolean complicated_logic
+    = (
+        if
+            !(
+                a
+                && b
+                || c
+                && (
+                    !d
+                    || !e
+                )
+                == (
+                    if
+                        foobar
+                    then
+                        gak
+                    else
+                        caz
+                )
+            )
+        then
+            "wow"
+        else
+            "WOWOWOW"
+    )
+```
+
+### `input_spacing`
+
+When making calls from a workflow, it is more readable and easier to edit if the supplied inputs are each on their own line. This does inflate the line count of a WDL dcoument, but it is worth it for the consistent readability. An exception _can_ be made (but does not have to be made), for calls with only a single parameter. In those cases, it is permissable to keep the input on the same line as the call.
+
+When newline separating inputs to a call, always follow each input with a trailing comma. See examples below.
+
+The assignment operator (`=`) should always be surrounded by whitespace.
+
+The `input:` token should always be on the same line and separated by one space from the opening curly bracket (`{`), likeso: `call foo_task { input:`. Never newline-separate the opening bracket (`{`) from the `input:` token.
+
+If there is a singular input and it can fit on that same line while being equal to or under the 90 character width limit, all tokens should be separated by a single space, likeso: `call foo_task { input: bam = sorted_bam }`. Note that there is no trailing comma here. This could optionally be split into 3 lines, likeso:
+
+```wdl
+call foo_task { input:
+    bam = sorted_bam,
+}
+```
+
+Note the trailing comma.
+
+**Group**: `spacing`
+
+#### Example
+
+Good (note that these examples take advantage of the input simplifying mechanism made available in the [v1.1 specification](https://github.com/openwdl/wdl/blob/main/versions/1.1/SPEC.md#call-statement)):
+
+```wdl
+call md5sum.compute_checksum { input: file = bam }
+
+call samtools.quickcheck { input: bam }
+
+call util.compression_integrity { input:
+    bgzipped_file = bam,
+}
+
+call markdups_post_wf.markdups_post { input:
+    markdups_bam = select_first([
+        markdups.duplicate_marked_bam,
+        "undefined",
+    ]),
+    markdups_bam_index = select_first([
+        markdups.duplicate_marked_bam_index,
+        "undefined",
+    ]),
+    coverage_beds,
+    coverage_labels = parse_input.labels,
+    prefix = post_subsample_prefix + ".MarkDuplicates",
+}
+```
+
+### `snake_case`
```

Comment:
- (Strong opinion) Disagree with this lint generally. Workflows and tasks should be PascalCase (e.g. `MyWorkflow` and `MyTask`), but variable names should be `snake_case`. The same as Python classes vs variables. I agree with structs as PascalCase as well.

Reply:
Interesting that you say `same as Python classes vs variables`. I view workflows and tasks to be analogous to Python **functions** not classes (structs would be most similar to classes). And of course Python functions are snake_cased like variables.

I think all casing choices are more-or-less equivalent. I propose leaving it up to some democratic method to determine the `sprocket` default.

Reply:
Yeah, I think of a `task` sort of like an object that has different attributes: a command block, runtime parameters, meta data, and also takes input/output info of course. I can see why it could feel either like a function or like an object.